### PR TITLE
Make open writers limit configurable in iceberg connector

### PR DESCRIPTION
**Pull request by akshayrai**
Addresses #6650

### CODE REVIEW

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergConfig.java [hunk 1]
```diff
@@ -74,4 +77,18 @@ public IcebergConfig setUseFileSizeFromMetadata(boolean useFileSizeFromMetadata)
...
         return this;
     }
+
+    @Min(1)
+    public int getMaxPartitionsPerWriter()
+    {
+        return maxPartitionsPerWriter;
+    }
+
+    @Config("iceberg.max-partitions-per-writers")
```
Comment (phd3) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergConfig.java:87]: nit: `writers` -->`writer`? I think it'd be okay to diverge from hive connector's config for in interest of better naming. @electrum any preference?

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSinkProvider.java [hunk 2]
```diff
@@ -39,18 +39,21 @@
...
             IcebergFileWriterFactory fileWriterFactory,
-            PageIndexerFactory pageIndexerFactory)
+            PageIndexerFactory pageIndexerFactory,
+            IcebergConfig config)
     {
         this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
         this.jsonCodec = requireNonNull(jsonCodec, "jsonCodec is null");
         this.fileWriterFactory = requireNonNull(fileWriterFactory, "fileWriterFactory is null");
         this.pageIndexerFactory = requireNonNull(pageIndexerFactory, "pageIndexerFactory is null");
+        this.maxOpenPartitions = config.getMaxPartitionsPerWriter();
```
Comment (phd3) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSinkProvider.java:56]: nit: add requireNonNull for config

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSink.java [hunk 3]
```diff
@@ -82,7 +82,7 @@
     private static final int MAX_PAGE_POSITIONS = 4096;
 
     @SuppressWarnings({"FieldCanBeLocal", "FieldMayBeStatic"})
```
Comment (phd3) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSink.java:84]: I guess this can be removed now

**Review by phd3 — COMMENTED** [review]
Thanks for the contribution, some minor comments. 

nit: I'd rename the commit message to "Make open writers limit configurable in iceberg connector"

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergConfig.java (hunk 1 above)
Comment (akshayrai) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergConfig.java]: updated to `iceberg.max-partitions-per-writer`

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSink.java (hunk 3 above)
Comment (akshayrai) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSink.java]: removed

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSinkProvider.java (hunk 2 above)
Comment (akshayrai) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSinkProvider.java:57]: added the check

**Review by phd3 — APPROVED** [review]
Looks good! can you please squash the fixup commit and update message?

**Comment by phd3** [comment]
Merged, thanks @akshayrai !

**Pull request merged**